src/components/Autocomplete/SectionItem/SectionItemText.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { Item } from '../../../types';
+import { isInGroupSuggestion } from '../../../typeGuards';
+import { escapeRegExp } from '../../../utils';
+
+export interface SectionItemTextProps {
+  item: Item;
+  query: string;
+  highlightSearchTerm?: boolean;
+}
+
+export default function SectionItemText({
+  item,
+  query,
+  highlightSearchTerm,
+}: SectionItemTextProps) {
+  const itemText = isInGroupSuggestion(item) ? item.groupName : item.value;
+
+  if (highlightSearchTerm) {
+    const queryRegex = new RegExp(`(${escapeRegExp(query)})`, 'gi');
+    const queryIncludedInText = queryRegex.test(itemText);
+
+    if (query && queryIncludedInText) {
+      const splitText = itemText.split(queryRegex);
+
+      return (
+        <>
+          {splitText[0]}
+          <b>{splitText[1]}</b>
+          {splitText[2]}
```

Comment:
There is a bug here, this will cut off the term if the query exists multiple times in the suggestion.

Reply:
@stanlp1 Yes you're right, fixed that!

The reason why I'm using a regex is that we want to get all query occurrences case insensitive and replace it with the actual case that was in the original string.

I guess this is going to be tricky using the query alone.

Reply:
👍 That makes sense.